contracts/voting-verifier/src/contract.rs

```diff
@@ -64,3 +65,781 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+                },
+            )
+            .unwrap(),
+        )
+        .unwrap();
+        assert_eq!(
+            statuses,
+            messages
+                .iter()
+                .map(|message| (message.cc_id.clone(), VerificationStatus::InProgress))
+                .collect::<Vec<(_, _)>>()
+        );
+
+        let msg = ExecuteMsg::Vote {
+            poll_id: Uint64::one().into(),
+            votes: (0..messages.len())
+                .map(|i| {
+                    if i % 2 == 0 {
+                        Vote::SucceededOnChain
+                    } else {
+                        Vote::NotFound
+                    }
+                })
+                .collect::<Vec<_>>(),
+        };
+
+        workers().iter().for_each(|worker| {
+            execute(
+                deps.as_mut(),
+                mock_env(),
+                mock_info(worker.address.as_str(), &[]),
+                msg.clone(),
+            )
+            .unwrap();
+        });
+
+        let msg = ExecuteMsg::EndPoll {
+            poll_id: Uint64::one().into(),
+        };
+
+        execute(
+            deps.as_mut(),
+            mock_env_expired(),
+            mock_info(SENDER, &[]),
+            msg,
+        )
+        .unwrap();
+
+        let statuses: Vec<(CrossChainId, VerificationStatus)> = from_binary(
+            &query(
+                deps.as_ref(),
+                mock_env(),
+                QueryMsg::GetMessagesStatus {
+                    messages: messages.clone(),
+                },
+            )
+            .unwrap(),
+        )
+        .unwrap();
+
+        assert_eq!(
+            statuses,
+            messages
+                .iter()
+                .enumerate()
+                .map(|(i, message)| (
+                    message.cc_id.clone(),
+                    if i % 2 == 0 {
+                        VerificationStatus::SucceededOnChain
+                    } else {
+                        VerificationStatus::NotFound
+                    }
+                ))
+                .collect::<Vec<(_, _)>>()
+        );
+    }
+
+    #[test]
+    fn should_start_worker_set_confirmation() {
+        let mut deps = setup();
+
+        let operators = Operators::new(vec![(vec![0, 1, 0, 1].into(), 1u64.into())], 1u64.into());
+        let msg = ExecuteMsg::VerifyWorkerSet {
+            message_id: message_id("id", 0),
+            new_operators: operators.clone(),
+        };
+        let res = execute(deps.as_mut(), mock_env(), mock_info(SENDER, &[]), msg);
+        assert!(res.is_ok());
+
+        let res: VerificationStatus = from_binary(
+            &query(
+                deps.as_ref(),
+                mock_env(),
+                QueryMsg::GetWorkerSetStatus {
+                    new_operators: operators.clone(),
+                },
+            )
+            .unwrap(),
+        )
+        .unwrap();
+        assert_eq!(res, VerificationStatus::InProgress);
```

Comment:
why does worker set verification return `InProgress` and messages return `None`? Is that an inconsistency?

Reply:
see my comment here: https://github.com/axelarnetwork/axelar-amplifier/pull/280#discussion_r1503443770